docs-ref-services/hdinsight.md

```diff
@@ -1,29 +1,391 @@
+  }
+}
+```
+
+If you're not logged into the correct subscription, select the correct one by running: 
+```azurecli-interactive
+az account set -s <name or ID of subscription>
+```
+
+> [!IMPORTANT]
+> If you have not already registered the HDInsight Resource Provider by another method (such as by creating an HDInsight Cluster through the Azure Portal), you need to do this once before you can authenticate. This can be done from the [Azure Cloud Shell](https://shell.azure.com/bash) by running the following command:
+>```azurecli-interactive
+>az provider register --namespace Microsoft.HDInsight
+>```
+
+Next, choose a name for your service principal and create it with the following command:
+
+```azurecli-interactive
+az ad sp create-for-rbac --name <Service Principal Name> --sdk-auth
+```
+
+The service principal information is displayed as JSON.
+
+```json
+{
+  "clientId": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+  "clientSecret": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+  "subscriptionId": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+  "tenantId": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
+  "activeDirectoryEndpointUrl": "https://login.microsoftonline.com",
+  "resourceManagerEndpointUrl": "https://management.azure.com/",
+  "activeDirectoryGraphResourceId": "https://graph.windows.net/",
+  "sqlManagementEndpointUrl": "https://management.core.windows.net:8443/",
+  "galleryEndpointUrl": "https://gallery.azure.com/",
+  "managementEndpointUrl": "https://management.core.windows.net/"
+}
+```
+Copy the below snippet and fill in `TENANT_ID`, `CLIENT_ID`, `CLIENT_SECRET`, and `SUBSCRIPTION_ID` with the strings from the JSON that was returned after running the command to create the service principal.
+
+```java
+import com.microsoft.azure.management.hdinsight.v2018_06_01_preview.*;
+import com.microsoft.azure.management.hdinsight.v2018_06_01_preview.implementation.HDInsightManagementClientImpl;
+
+public class Main {
+    public static void main (String[] args) {
+
+        // Tenant ID for your Azure Subscription
+        String TENANT_ID = "";
+        // Your Service Principal App Client ID
+        String CLIENT_ID = "";
+        // Your Service Principal Client Secret
+        String CLIENT_SECRET = "";
+        // Azure Subscription ID
+        String SUBSCRIPTION_ID = "";
+
+        ApplicationTokenCredentials credentials = new ApplicationTokenCredentials(
+                CLIENT_ID,
+                TENANT_ID,
+                CLIENT_SECRET,
+                AzureEnvironment.AZURE);
+
+        HDInsightManagementClientImpl client = new HDInsightManagementClientImpl(credentials);
+```
+
+
+## Cluster Management
+> [!NOTE]
+> This section assumes you have already authenticated and constructed an `HDInsightManagementClientImpl` instance and store it in a variable called `client`. Instructions for authenticating and obtaining an `HDInsightManagementClientImpl` can be found in the Authentication section above.
+
+### Create a Cluster
```

Comment:
It is best practice to keep elements in a Markdown with blank lines over and below, especially the titles.

As in: 

```

# Title

Text
```

Reply:
Done, thanks